File: frontend/src/types/Booking.ts
export interface Booking {
    id: number;
    sessionId: number;
    patientId: string;
    patientName?: string;
    patient?: {
        id?: string;
        firstName?: string;
        lastName?: string;
        email?: string;
        phoneNumber?: string;
        contactNumbers?: string[];
    };
    session?: {
        id: number;
        doctorId: string;
        doctorName?: string;
        date: string;
        startTime: string;
        endTime: string;
        capacity: number;
        sessionFee: number;
        description: string;
        canceled: boolean;
        completed: boolean;
        ongoing?: boolean;
    };
    bookedDateandTime: string; // ISO date string
    positionInQueue?: number;
    onGoing: boolean;
    completed: boolean;
}

File: backend/Models/Booking.cs
using System.ComponentModel.DataAnnotations;

namespace backend.Models
{

    public class Booking
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int SessionId { get; set; }

        public Session Session { get; set; } = null!;

        [Required]
        public string PatientId { get; set; } = null!;

        public ApplicationUser Patient { get; set; } = null!;

        [Required]
        public DateTime BookedDateandTime { get; set; }

        public int positionInQueue { get; set; }

        public bool OnGoing { get; set; } = false;

        public bool Completed { get; set; } = false;

    }
};